### ISSUE TITLE
Cannot specify both EBS volume and no-device for block devices

### ISSUE THREAD

**Issue by Pankajc123** [aws/aws-cdk#5922]
I am getting error "Cannot specify both EBS volume and no-device for block devices" in CloudFormation stack while creating autoscaling group.

### Reproduction Steps

1. Create autoscaling group using code snippet below.

```
var autoscalinggropup = new AutoScalingGroup(this, "AutoScalingGroup", new AutoScalingGroupProps{
                Cooldown = Duration.Seconds(300),
                MaxCapacity = additionalinstances,
                MinCapacity = additionalinstances,
                HealthCheck = HealthCheck.Ec2(new Ec2HealthCheckOptions{
                    Grace = Duration.Seconds(3600)
                }),
                Vpc = vpc,
                VpcSubnets = selectedsubnet,
                Role = ec2iamrole,
                InstanceType = new InstanceType(instancetypeval),
                MachineImage = new GenericLinuxImage(amiDictionary),
                BlockDevices = new[]{blockdeviceautoscaling},
                KeyName = keypairname,
                UpdateType = UpdateType.REPLACING_UPDATE                
            });                       
            autoscalinggropup.AddSecurityGroup(securitygroup);

```

2. Run "cdk synth" command. It will generate CF template for above resource below.
```
"AutoScalingGroupLaunchConfigDEEB160C": {
      "Type": "AWS::AutoScaling::LaunchConfiguration",
      "Properties": {
        "ImageId": "ami-08fd8ae3806f09a08",
        "InstanceType": "m5.2xlarge",
        "BlockDeviceMappings": [
          {
            "DeviceName": "/dev/sda1",
            "Ebs": {
              "DeleteOnTermination": true,
              "VolumeSize": 200,
              "VolumeType": "standard"
            },
            "NoDevice": false
          }
        ],
        "IamInstanceProfile": {
          "Ref": "AutoScalingGroupInstanceProfile342FAC7C"
        },
        "KeyName": "knew1",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "AutoScalingGroupInstanceSecurityGroup9D2E0C5E",
              "GroupId"
            ]
          },
          {
            "Fn::GetAtt": [
              "SecurityGroupDD263621",
              "GroupId"
            ]
          }
        ],
        "UserData": {
          "Fn::Base64": "#!/bin/bash"
        }
      },
      "DependsOn": [
        "IAMRole24D9329E"
      ],
      "Metadata": {
        "aws:cdk:path": "SamplesStack/AutoScalingGroup/LaunchConfig"
      }
    },
    "AutoScalingGroupASG804C35BE": {
      "Type": "AWS::AutoScaling::AutoScalingGroup",
      "Properties": {
        "MaxSize": "2",
        "MinSize": "2",
        "Cooldown": "300",
        "HealthCheckGracePeriod": 3600,
        "HealthCheckType": "EC2",
        "LaunchConfigurationName": {
          "Ref": "AutoScalingGroupLaunchConfigDEEB160C"
        },
        "Tags": [
          {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "SamplesStack/AutoScalingGroup"
          }
        ],
        "VPCZoneIdentifier": [
          "subnet-0c4610e4be972bcb5"
        ]
      },
      "UpdatePolicy": {
        "AutoScalingReplacingUpdate": {
          "WillReplace": true
        },
        "AutoScalingScheduledAction": {
          "IgnoreUnmodifiedGroupSizeProperties": true
        }
      },
      "Metadata": {
        "aws:cdk:path": "SamplesStack/AutoScalingGroup/ASG"
      }
    }

```

3. Run "cdk deploy" command. It starts creating CF stack and then it's failed to create because of error "Cannot specify both EBS volume and no-device for block devices"

When I compile CF template, not sure why it's embedding **""NoDevice": false"** property for block devices mapping inside launch configuration resource.

### Error Log
"Cannot specify both EBS volume and no-device for block devices"

### Environment

  - **CLI Version      :*1.21.1*
  - **Framework Version:**
  - **OS               :*Windows*
  - **Language         :*.Net*

### Other

I am logging this as "bug" because i have seen few users have discussed this in "gitter".
Can you please suggest workaround for this problem?

--- 

This is :bug: Bug Report

**Comment by lrhazi-cua** [comment]
I get the same error using Python:

```
> cdk --version
1.20.0 (build 021c521)
```

```
asg = autoscaling.AutoScalingGroup(
            self,
            "ASG",
            vpc=vpc,
            instance_type=ec2.InstanceType.of(
                ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO
            ),
            machine_image=ec2.GenericLinuxImage(
                props['webserver']['ami']),
            user_data=inst_data,
            key_name=props['webserver']['ssh_key'],
            block_devices=[
                autoscaling.BlockDevice(
                    device_name="/dev/sda1", volume=autoscaling.BlockDeviceVolume.ebs(10)),
                autoscaling.BlockDevice(
                    device_name="/dev/xvdb", volume=autoscaling.BlockDeviceVolume.ebs(40)),
            ],
        )
```

```
toptier-stack: deploying...
toptier-stack: creating CloudFormation changeset...
 0/4 | 2:06:38 AM | UPDATE_IN_PROGRESS   | AWS::AutoScaling::LaunchConfiguration     | ASG/LaunchConfig (ASGLaunchConfigC00AF12B) Requested update requires the creation of a new physical resource; hence creating one.
 1/4 | 2:06:38 AM | UPDATE_FAILED        | AWS::AutoScaling::LaunchConfiguration     | ASG/LaunchConfig (ASGLaunchConfigC00AF12B) Cannot specify both EBS volume and no-device for block devices. (Service: AmazonAutoScaling; Status Code: 400; Error Code: ValidationError; Request ID: 8fd305dc-40d3-11ea-a71f-5bb65e24afaa)
        new AutoScalingGroup (C:\Users\lrhazi\AppData\Local\Temp\jsii-kernel-fGjRCZ\node_modules\@aws-cdk\aws-autoscaling\lib\auto-scaling-group.js:144:30)
        \_ C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7762:49
        \_ Kernel._wrapSandboxCode (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:8222:20)
        \_ Kernel._create (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7762:26)
        \_ Kernel.create (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7509:21)
        \_ KernelHost.processRequest (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7296:28)
        \_ KernelHost.run (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7236:14)
        \_ Immediate._onImmediate (C:\Users\lrhazi\Documents\CodingProjects\aws-toptier-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:7239:37)
        \_ processImmediate (internal/timers.js:439:21)
```

**Comment by Pankajc123** [comment]
I found that, this is fixed in CDK version 1.24.0. Just fyi.

**Comment by SomayaB** [comment]
Closing this issue since it looks like it's been resolved. Feel free to reopen if you're still experiencing it.
